modules/gcp/gcr.go

```diff
@@ -0,0 +1,89 @@
+package gcp
+
+import (
+	"fmt"
+
+	"testing"
+
+	gcrname "github.com/google/go-containerregistry/pkg/name"
+	gcrgoogle "github.com/google/go-containerregistry/pkg/v1/google"
+	gcrremote "github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/gruntwork-io/terratest/modules/logger"
+)
+
+// DeleteGCRRepo deletes a GCR repository including all tagged images
+func DeleteGCRRepo(t *testing.T, repo string) {
+	err := DeleteGCRRepoE(t, repo)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// DeleteGCRRepoE deletes a GCR repository including all tagged images
+func DeleteGCRRepoE(t *testing.T, repo string) error {
+	// create a new auther for the API calls
+	auther, err := gcrgoogle.NewEnvAuthenticator()
+	if err != nil {
+		return fmt.Errorf("Failed to create auther. Got error: %v", err)
+	}
+
+	gcrrepo, err := gcrname.NewRepository(repo)
+	if err != nil {
+		return fmt.Errorf("Failed to get repo. Got error: %v", err)
+	}
+
+	logger.Logf(t, "Retriving Image Digests %s", gcrrepo)
+	tags, err := gcrgoogle.List(gcrrepo, gcrgoogle.WithAuth(auther))
+	if err != nil {
+		return fmt.Errorf("Failed to list tags for repo %s. Got error: %v", repo, err)
+	}
+
+	// attempt to delete the latest image tag
+	latestRef := repo + ":latest"
+	logger.Logf(t, "Deleting Image Ref %s", latestRef)
+	if err := DeleteGCRImageRefE(t, latestRef); err != nil {
+		return fmt.Errorf("Failed to delete GCR Image Reference %s. Got error: %v", latestRef, err)
+	}
+
+	// delete image references sequentially
+	for k := range tags.Manifests {
+		ref := repo + "@" + k
+		logger.Logf(t, "Deleting Image Ref %s", ref)
+
+		if err := DeleteGCRImageRefE(t, ref); err != nil {
+			return fmt.Errorf("Failed to delete GCR Image Reference %s. Got error: %v", ref, err)
+		}
+	}
+
+	return nil
+}
+
+// DeleteGCRImageRef deletes a single repo image ref/digest
+func DeleteGCRImageRef(t *testing.T, ref string) {
+	err := DeleteGCRImageRefE(t, ref)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// DeleteGCRImageRefE deletes a single repo image ref/digest
+func DeleteGCRImageRefE(t *testing.T, ref string) error {
+	name, err := gcrname.ParseReference(ref)
+	if err != nil {
+		return fmt.Errorf("Failed to parse reference %s. Got error: %v", ref, err)
+	}
+
+	// create a new auther for the API calls
+	auther, err := gcrgoogle.NewEnvAuthenticator()
+	if err != nil {
+		return fmt.Errorf("Failed to create auther. Got error: %v", err)
+	}
+
+	opts := gcrremote.WithAuth(auther)
+
+	if err := gcrremote.Delete(name, opts); err != nil {
+		return fmt.Errorf("Failed to delete %s. Got error: %v", name, err)
+	}
+
+	return nil
+}
```

Comment:
I am not sure these are interfaces we want to maintain, as they fail the litmus test for "Creating infrastructure": https://github.com/gruntwork-io/terratest/blob/master/CONTRIBUTING.md#types-of-contributions

Is there a reason you can't use a terraform example for the GCR repositories?

Reply:
After reading through https://stackoverflow.com/questions/53714284/how-do-you-create-a-gcr-repo-with-terraform, this makes sense to me given that there is no way to explicitly manage GCR repositories on the fly using `terraform`. This means that to properly clean up a test, you have to manually delete the images that are pushed using `docker`, and these functions makes it easy to do that.

Reply:
Ok